src/settings.cpp

```diff
@@ -1262,6 +1263,20 @@ void LoadFromConfig(bool startup)
 			}
 		}
 
+		/* Persist the right click close option from older versions. */
+		if (generic_version < IFV_RIGHT_CLICK_CLOSE) {
+			IniGroup *gui = generic_ini.GetGroup("gui", false);
+			if (gui != nullptr) {
+				IniItem *right_mouse_wnd_close = gui->GetItem("right_mouse_wnd_close");
+				if (right_mouse_wnd_close != nullptr) {
+					if (right_mouse_wnd_close->value) {
+						std::string str = right_mouse_wnd_close->value.value();
+						_settings_client.gui.right_click_wnd_close = strcmp(str.c_str(), "true") == 0 ? 1 : 0;
+					}
```

Comment:
We should reusing setting code here. Will write some code for that tomorrow. This are reminder to myself :)

Reply:
I am figure out with other developers what we exactly want here, and it seems we are narrowing down on a solution. Hang tight, I will help out with this soon :)

Reply:
After a rebase, you can do something like:

```
		if (generic_version < IFV_RIGHT_CLICK_CLOSE && IsConversionNeeded(generic_ini, "gui", "right_mouse_wnd_close", "right_click_wnd_close", &old_item)) {
			auto old_value = BoolSettingDesc::ParseSingleValue(old_item->value->c_str());
			_settings_client.gui.right_click_wnd_close = old_value.value_or(false) ? RCC_YES : RCC_NO;
		}
```

Ignoring any typos I might have made :D